## 3. Perguntas antes do código

A aula enfatiza que a maior armadilha é começar com `groupby()` sem saber o que se quer responder. fileciteturn5file0

### Tarefa
Antes de programar, escreva em markdown respostas curtas para estas perguntas:

1. Qual canal de venda é o mais eficiente?
2. Qual categoria parece mais rentável?
3. Quais produtos são os campeões por lucro?
4. O negócio está crescendo ao longo do tempo?
5. Quais KPIs globais resumem melhor a operação?

Depois disso, avance para o código.


# Atividade Prática — Aula 4: Análise de Dados e Construção de Indicadores (KPIs)

Esta atividade foi construída com base na Aula 4, que marca a transição **do dado limpo para a tomada de decisão estratégica**, com foco em **KPIs, métricas derivadas, dimensões de análise, rankings e insights executivos**. fileciteturn5file0

## Ideia central da aula
Antes do código, vem a **pergunta de negócio**.  
Depois, definimos:
1. a **métrica**
2. a **dimensão**
3. a **agregação**
4. e só então construímos a resposta em tabela e em texto. fileciteturn5file0

## Regras da atividade
- O notebook **orienta**, mas você deve **escrever os códigos**.
- Sempre que gerar uma tabela agregada, escreva **1 ou 2 frases interpretando o resultado**.
- O objetivo não é apenas calcular números, mas comunicar decisões.

## Dataset da atividade
Arquivo: `vendas_brasil_aula4_kpis.csv`


## 1. Preparação do ambiente

Importe as bibliotecas necessárias para a atividade.

**Sugestão:**  
- `pandas`
- `numpy`

Se desejar, você também pode usar uma biblioteca de visualização mais adiante, mas o foco desta aula é **tabela agregada + insight**.


In [5]:
import pandas as pd
import numpy as np

## 2. Leitura da base

Leia o arquivo `vendas_brasil_aula4_kpis.csv` em um DataFrame chamado `df`.

Depois:
1. exiba as primeiras linhas
2. veja o tamanho da base
3. inspecione os tipos das colunas
4. confirme se `data_venda` está pronta para análises temporais


In [6]:
df = pd.read_csv('vendas_brasil_aula4_kpis.csv')

print('Primeiras linhas do DataFrame:')
print(df.head())

print('\nTamanho da base (linhas, colunas):')
print(df.shape)

print('\nTipos das colunas:')
print(df.info())

print('\nVerificando o tipo da coluna data_venda:')
print(df['data_venda'].dtype)

FileNotFoundError: [Errno 2] No such file or directory: 'vendas_brasil_aula4_kpis.csv'

## 3. Perguntas antes do código

A aula enfatiza que a maior armadilha é começar com `groupby()` sem saber o que se quer responder. fileciteturn5file0

### Tarefa
Antes de programar, escreva em markdown respostas curtas para estas perguntas:

1.  Qual canal de venda é o mais eficiente?
    *   **R:** Precisamos calcular a margem de lucro por canal para determinar a eficiência.
2.  Qual categoria parece mais rentável?
    *   **R:** Será necessário analisar o lucro total e a margem por categoria.
3.  Quais produtos são os campeões por lucro?
    *   **R:** Faremos um ranking dos produtos com maior lucro.
4.  O negócio está crescendo ao longo do tempo?
    *   **R:** A análise temporal da receita e do lucro por período (mês/trimestre) responderá a essa questão.
5.  Quais KPIs globais resumem melhor a operação?
    *   **R:** Receita Total, Lucro Total, Margem de Lucro Global e Ticket Médio são os KPIs globais essenciais.

Depois disso, avance para o código.

## 4. KPIs globais da operação

Com base nas “4 Métricas de Ouro do Varejo” apresentadas na aula, calcule os seguintes KPIs globais da base inteira: fileciteturn5file0

- **Receita Total**
- **Lucro Total**
- **Margem de Lucro (%)**
- **Ticket Médio**

### Orientação
- Pense primeiro na fórmula de cada KPI
- Escreva os cálculos em células separadas ou em uma pequena tabela-resumo
- Se quiser, formate os resultados para facilitar a leitura


In [ ]:
receita_total = df['receita'].sum()
lucro_total = df['lucro'].sum()
margem_lucro_percentual = (lucro_total / receita_total) * 100
ticket_medio = receita_total / df['pedido_id'].nunique()

print(f"Receita Total: R$ {receita_total:,.2f}")
print(f"Lucro Total: R$ {lucro_total:,.2f}")
print(f"Margem de Lucro: {margem_lucro_percentual:.2f}%")
print(f"Ticket Médio: R$ {ticket_medio:,.2f}")

Receita Total: R$ 1,921,492.22
Lucro Total: R$ 544,496.15
Margem de Lucro: 28.34%
Ticket Médio: R$ 4,574.98


### Interpretação
Escreva 3 a 5 linhas respondendo:
- O negócio parece grande ou pequeno?
- O lucro acompanha bem a receita?
- O ticket médio sugere compras de maior valor ou de menor valor?

**R:**
O negócio demonstra ser considerável. Margem de Lucro de 28.34% indicam que o lucro acompanha a receita de forma saudável, sugerindo uma boa gestão de custos ou boa precificação. O Ticket Médio de R$ 4,574.98 sugere que as vendas envolvem produtos de valor mais elevado, o que pode indicar um foco em clientes que realizam compras mais significativas ou produtos de maior custo unitário.

## 5. Métricas derivadas

A aula mostra que **somar valores não basta**.  
Precisamos criar métricas derivadas para avaliar eficiência e comportamento. fileciteturn5file0

### Tarefa
Crie, quando fizer sentido:
- uma coluna de **margem_lucro** (`lucro / receita`)
- uma lógica para **ticket_medio** da operação ou por grupo

### Reflexão
Explique:
1. Por que receita alta não significa automaticamente melhor desempenho?
2. Em que situação a margem ajuda mais do que a receita bruta?

**R:**
1.
    Receita alta pode ser enganosa, pois não considera os custos associados à sua geração. Um negócio com alta receita, mas com custos operacionais ainda mais altos, pode resultar em baixo ou até nenhum lucro. O desempenho real de um negócio é medido pela sua capacidade de gerar lucro e valor, não apenas pelo volume de vendas.<br><br>
2.
    A margem é crucial em situações onde a eficiência e a lucratividade são prioritárias. Ela ajuda a comparar a rentabilidade de diferentes produtos, canais ou categorias, mesmo que tenham receitas muito distintas. Por exemplo, um produto com receita bruta menor, mas com uma margem de lucro significativamente maior, pode ser mais valioso para o negócio do que um produto de alta receita com margem baixa. A margem é essencial para decisões de precificação, controle de custos e alocação de recursos.

In [ ]:
df['margem_lucro'] = df['lucro'] / df['receita']
print("Coluna 'margem_lucro' criada. Primeiras 5 linhas com a nova coluna:")
print(df[['receita', 'lucro', 'margem_lucro']].head())

Coluna 'margem_lucro' criada. Primeiras 5 linhas com a nova coluna:
    receita    lucro  margem_lucro
0    475.30   239.08      0.503009
1   5727.16  1769.48      0.308963
2   7231.10  2449.85      0.338794
3   2798.88  1356.67      0.484719
4  12539.40  3656.84      0.291628


## 6. Fatiando a realidade: análise por dimensão

A aula mostra que um KPI isolado não decide nada.  
É preciso perguntar: **como isso varia por canal, categoria, região e tempo?** fileciteturn5file0

### Tarefa
Liste, em markdown, quais colunas do dataset representam:
- métrica
- dimensão
- tempo

**R:**
- **Métrica**: `receita`, `lucro`, `margem_lucro`, `quantidade_pedidos`, `preco_unitario`
- **Dimensão**: `uf`, `canal_venda`, `categoria`, `produto`
- **Tempo**: `data_venda`

## 7. Análise por canal

### Pergunta de negócio
**Qual canal de venda é o mais eficiente?**

### Tarefa
Crie uma tabela agregada por `canal_venda` contendo, no mínimo:
- Receita Total
- Lucro Total
- Margem Média ou Margem Calculada
- Quantidade de pedidos
- Ticket Médio

### Depois responda
1. Qual canal gera mais receita?
2. Qual canal gera maior eficiência?
3. Se você fosse gestor, priorizaria volume ou margem?


In [ ]:
analise_canal = df.groupby('canal_venda').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum'),
    quantidade_pedidos=('pedido_id', 'nunique')
).reset_index()

analise_canal['margem_calculada'] = (analise_canal['lucro_total'] / analise_canal['receita_total']) * 100
analise_canal['ticket_medio'] = analise_canal['receita_total'] / analise_canal['quantidade_pedidos']

analise_canal['receita_total'] = analise_canal['receita_total'].map('R$ {:,.2f}'.format)
analise_canal['lucro_total'] = analise_canal['lucro_total'].map('R$ {:,.2f}'.format)
analise_canal['margem_calculada'] = analise_canal['margem_calculada'].map('{:,.2f}%'.format)
analise_canal['ticket_medio'] = analise_canal['ticket_medio'].map('R$ {:,.2f}'.format)

display(analise_canal.sort_values(by='receita_total', ascending=False))

,canal_venda,receita_total,lucro_total,quantidade_pedidos,margem_calculada,ticket_medio
1,Marketplace,"R$ 657,802.09","R$ 185,802.57",141,28.25%,"R$ 4,665.26"
0,Loja Física,"R$ 640,797.06","R$ 174,583.16",143,27.24%,"R$ 4,481.10"
2,Online,"R$ 622,893.07","R$ 184,110.42",136,29.56%,"R$ 4,580.10"


### Insight obrigatório
Escreva 1 ou 2 frases interpretando sua tabela, no estilo executivo:
- não apenas “o Online foi maior”
- mas o que isso **significa para a decisão**

**R:** O canal Marketplace é o maior em receita e lucro absoluto, indicando volume de vendas. Mas o canal de vendas Online teve maior margem de lucro.

## 8. Análise por categoria

### Pergunta de negócio
**Qual categoria é mais lucrativa e qual parece mais eficiente?**

### Tarefa
Crie uma tabela agregada por `categoria` com:
- Receita Total
- Lucro Total
- Margem
- Ticket Médio

### Depois responda
1. A categoria de maior receita também é a de maior margem?
2. Existe alguma categoria que parece vender bem, mas ser pouco eficiente?


In [ ]:
analise_categoria = df.groupby('categoria').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum'),
    quantidade_pedidos=('pedido_id', 'nunique')
).reset_index()

analise_categoria['margem_calculada'] = (analise_categoria['lucro_total'] / analise_categoria['receita_total']) * 100
analise_categoria['ticket_medio'] = analise_categoria['receita_total'] / analise_categoria['quantidade_pedidos']

analise_categoria['receita_total'] = analise_categoria['receita_total'].map('R$ {:,.2f}'.format)
analise_categoria['lucro_total'] = analise_categoria['lucro_total'].map('R$ {:,.2f}'.format)
analise_categoria['margem_calculada'] = analise_categoria['margem_calculada'].map('{:,.2f}%'.format)
analise_categoria['ticket_medio'] = analise_categoria['ticket_medio'].map('R$ {:,.2f}'.format)

display(analise_categoria.sort_values(by='receita_total', ascending=False))

,categoria,receita_total,lucro_total,quantidade_pedidos,margem_calculada,ticket_medio
2,Telefonia,"R$ 866,292.45","R$ 219,975.58",109,25.39%,"R$ 7,947.64"
0,Informática,"R$ 805,376.44","R$ 219,090.15",93,27.20%,"R$ 8,659.96"
3,Áudio,"R$ 132,900.25","R$ 55,006.02",111,41.39%,"R$ 1,197.30"
1,Periféricos,"R$ 116,923.08","R$ 50,424.40",107,43.13%,"R$ 1,092.74"


## 10. O motor da análise: agregações (groupby)

Nos slides, a lógica central é:
- `receita -> sum`
- `margem -> mean` ou cálculo derivado
- `produto -> count`
fileciteturn5file0

### Tarefa
Escolha **duas dimensões diferentes** e construa tabelas agregadas mostrando que você entendeu essa lógica.

Exemplos:
- por canal
- por categoriaa
- por UF
- por mês


In [ ]:
print("\n--- Análise por UF ---")
analise_uf = df.groupby('uf').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum'),
    quantidade_pedidos=('pedido_id', 'nunique')
).reset_index()

analise_uf['margem_calculada'] = (analise_uf['lucro_total'] / analise_uf['receita_total']) * 100
analise_uf['ticket_medio'] = analise_uf['receita_total'] / analise_uf['quantidade_pedidos']

analise_uf['receita_total'] = analise_uf['receita_total'].map('R$ {:,.2f}'.format)
analise_uf['lucro_total'] = analise_uf['lucro_total'].map('R$ {:,.2f}'.format)
analise_uf['margem_calculada'] = analise_uf['margem_calculada'].map('{:,.2f}%'.format)
analise_uf['ticket_medio'] = analise_uf['ticket_medio'].map('R$ {:,.2f}'.format)

display(analise_uf.sort_values(by='receita_total', ascending=False))

print("\n--- Análise por Produto ---")
analise_produto = df.groupby('produto').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum')
).reset_index()

analise_produto['margem_calculada'] = (analise_produto['lucro_total'] / analise_produto['receita_total']) * 100

analise_produto['receita_total'] = analise_produto['receita_total'].map('R$ {:,.2f}'.format)
analise_produto['lucro_total'] = analise_produto['lucro_total'].map('R$ {:,.2f}'.format)
analise_produto['margem_calculada'] = analise_produto['margem_calculada'].map('{:,.2f}%'.format)

display(analise_produto.sort_values(by='receita_total', ascending=False).head(10)) # Mostrando apenas os Top 10 para conciseness


--- Análise por UF ---


,uf,receita_total,lucro_total,quantidade_pedidos,margem_calculada,ticket_medio
9,SP,"R$ 246,835.50","R$ 71,022.16",48,28.77%,"R$ 5,142.41"
6,RJ,"R$ 218,631.91","R$ 63,960.39",38,29.25%,"R$ 5,753.47"
4,PE,"R$ 208,457.44","R$ 58,183.95",55,27.91%,"R$ 3,790.14"
3,MG,"R$ 207,614.69","R$ 61,122.25",44,29.44%,"R$ 4,718.52"
8,SC,"R$ 198,143.47","R$ 60,272.32",40,30.42%,"R$ 4,953.59"
7,RS,"R$ 187,048.31","R$ 49,795.96",43,26.62%,"R$ 4,349.96"
1,ES,"R$ 181,113.70","R$ 49,838.07",39,27.52%,"R$ 4,643.94"
5,PR,"R$ 174,136.19","R$ 46,155.64",35,26.51%,"R$ 4,975.32"
0,BA,"R$ 157,082.99","R$ 42,117.30",42,26.81%,"R$ 3,740.07"
2,GO,"R$ 142,428.02","R$ 42,028.11",36,29.51%,"R$ 3,956.33"



--- Análise por Produto ---


,produto,receita_total,lucro_total,margem_calculada
7,Teclado Mecânico,"R$ 79,973.75","R$ 33,939.07",42.44%
0,Caixa de Som,"R$ 68,956.76","R$ 28,316.37",41.06%
1,Headset Pro,"R$ 63,943.49","R$ 26,689.65",41.74%
4,Notebook Pro,"R$ 594,132.74","R$ 153,094.84",25.77%
5,Smartphone X,"R$ 489,572.08","R$ 118,198.25",24.14%
6,Tablet Plus,"R$ 376,720.37","R$ 101,777.33",27.02%
3,Mouse Gamer,"R$ 36,949.33","R$ 16,485.33",44.62%
2,Monitor 27,"R$ 211,243.70","R$ 65,995.31",31.24%


## 11. Campeões e vilões (Top N)

A aula destaca o poder dos rankings curtos e focados. fileciteturn5file0

### Tarefa
Crie:
- Top 10 produtos por **lucro**
- Top 10 produtos por **receita**
- 5 categorias ou UFs com **pior margem** ou pior desempenho

### Perguntas
1. Quem são os “campeões”?
2. Quem são os “vilões”?
3. Que ação tática poderia ser tomada com base nisso?


In [ ]:
analise_produto_numeric = analise_produto.copy()
analise_produto_numeric['receita_total'] = analise_produto_numeric['receita_total'].str.replace('R$', '', regex=False).str.replace('.', '', regex=False).str.replace(',', '.', regex=False).astype(float)
analise_produto_numeric['lucro_total'] = analise_produto_numeric['lucro_total'].str.replace('R$', '', regex=False).str.replace('.', '', regex=False).str.replace(',', '.', regex=False).astype(float)
analise_produto_numeric['margem_calculada'] = analise_produto_numeric['margem_calculada'].str.replace('%', '', regex=False).str.replace(',', '.', regex=False).astype(float)

analise_categoria_numeric = analise_categoria.copy()
analise_categoria_numeric['margem_calculada'] = analise_categoria_numeric['margem_calculada'].str.replace('%', '', regex=False).str.replace(',', '.', regex=False).astype(float)

analise_uf_numeric = analise_uf.copy()
analise_uf_numeric['margem_calculada'] = analise_uf_numeric['margem_calculada'].str.replace('%', '', regex=False).str.replace(',', '.', regex=False).astype(float)

print("\n--- Top 10 produtos por lucro ---")
display(analise_produto_numeric.sort_values(by='lucro_total', ascending=False).head(10).style.format({
    'receita_total': 'R$ {:,.2f}',
    'lucro_total': 'R$ {:,.2f}',
    'margem_calculada': '{:,.2f}%'
}))

print("\n--- Top 10 produtos por receita ---")
display(analise_produto_numeric.sort_values(by='receita_total', ascending=False).head(10).style.format({
    'receita_total': 'R$ {:,.2f}',
    'lucro_total': 'R$ {:,.2f}',
    'margem_calculada': '{:,.2f}%'
}))

print("\n--- 5 Categorias com pior margem ---")
display(analise_categoria_numeric.sort_values(by='margem_calculada', ascending=True).head(5).style.format({
    'receita_total': 'R$ {:,.2f}',
    'lucro_total': 'R$ {:,.2f}',
    'margem_calculada': '{:,.2f}%',
    'ticket_medio': 'R$ {:,.2f}'
}))

print("\n--- 5 UFs com pior margem ---")
display(analise_uf_numeric.sort_values(by='margem_calculada', ascending=True).head(5).style.format({
    'receita_total': 'R$ {:,.2f}',
    'lucro_total': 'R$ {:,.2f}',
    'margem_calculada': '{:,.2f}%',
    'ticket_medio': 'R$ {:,.2f}'
}))



--- Top 10 produtos por lucro ---


,produto,receita_total,lucro_total,margem_calculada
4,Notebook Pro,R$ 594.13,R$ 153.09,25.77%
5,Smartphone X,R$ 489.57,R$ 118.20,24.14%
6,Tablet Plus,R$ 376.72,R$ 101.78,27.02%
2,Monitor 27,R$ 211.24,R$ 66.00,31.24%
7,Teclado Mecânico,R$ 79.97,R$ 33.94,42.44%
0,Caixa de Som,R$ 68.96,R$ 28.32,41.06%
1,Headset Pro,R$ 63.94,R$ 26.69,41.74%
3,Mouse Gamer,R$ 36.95,R$ 16.49,44.62%



--- Top 10 produtos por receita ---


,produto,receita_total,lucro_total,margem_calculada
4,Notebook Pro,R$ 594.13,R$ 153.09,25.77%
5,Smartphone X,R$ 489.57,R$ 118.20,24.14%
6,Tablet Plus,R$ 376.72,R$ 101.78,27.02%
2,Monitor 27,R$ 211.24,R$ 66.00,31.24%
7,Teclado Mecânico,R$ 79.97,R$ 33.94,42.44%
0,Caixa de Som,R$ 68.96,R$ 28.32,41.06%
1,Headset Pro,R$ 63.94,R$ 26.69,41.74%
3,Mouse Gamer,R$ 36.95,R$ 16.49,44.62%



--- 5 Categorias com pior margem ---


ValueError: Unknown format code 'f' for object of type 'str'


--- 5 UFs com pior margem ---


ValueError: Unknown format code 'f' for object of type 'str'

## 12. Análise temporal

A aula reforça que sem tempo a análise fica estática. fileciteturn5file0

### Tarefa
1. Converta `data_venda` para data, se necessário
2. Crie colunas auxiliares como:
   - mês
   - trimestre (opcional)
3. Gere uma tabela agregada por mês com:
   - receita
   - lucro
   - ticket médio ou margem

### Depois responda
1. O lucro parece crescer ao longo do tempo?
2. O último trimestre foi melhor que o anterior?
3. Quais meses concentram as vendas de fim de ano?

**R:** (Serão respondidas após a execução da célula de código.)

In [ ]:

df['data_venda'] = pd.to_datetime(df['data_venda'])


df['mes'] = df['data_venda'].dt.to_period('M')
df['trimestre'] = df['data_venda'].dt.to_period('Q')


analise_temporal_mensal = df.groupby('mes').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum')
).reset_index()

analise_temporal_mensal['margem_calculada'] = (analise_temporal_mensal['lucro_total'] / analise_temporal_mensal['receita_total']) * 100


analise_temporal_mensal['receita_total'] = analise_temporal_mensal['receita_total'].map('R$ {:,.2f}'.format)
analise_temporal_mensal['lucro_total'] = analise_temporal_mensal['lucro_total'].map('R$ {:,.2f}'.format)
analise_temporal_mensal['margem_calculada'] = analise_temporal_mensal['margem_calculada'].map('{:,.2f}%'.format)

print("\n--- Análise Temporal Mensal ---")
display(analise_temporal_mensal)


--- Análise Temporal Mensal ---


,mes,receita_total,lucro_total,margem_calculada
0,2025-01,"R$ 161,571.89","R$ 46,515.14",28.79%
1,2025-02,"R$ 108,587.03","R$ 29,998.87",27.63%
2,2025-03,"R$ 87,902.22","R$ 23,923.66",27.22%
3,2025-04,"R$ 148,930.46","R$ 43,094.00",28.94%
4,2025-05,"R$ 106,355.17","R$ 30,334.32",28.52%
5,2025-06,"R$ 134,685.83","R$ 38,592.70",28.65%
6,2025-07,"R$ 196,651.96","R$ 57,913.44",29.45%
7,2025-08,"R$ 160,240.62","R$ 45,881.36",28.63%
8,2025-09,"R$ 109,291.03","R$ 31,938.44",29.22%
9,2025-10,"R$ 158,226.19","R$ 41,463.02",26.20%


## 13. Fluxo de investigação do analista

A aula apresenta a sequência:
**Pergunta -> Colunas necessárias -> Tipo de resposta/gráfico**. fileciteturn5file0

### Tarefa
Preencha, em markdown, pelo menos 3 exemplos no formato:

- **Pergunta de negócio:** ...
- **Colunas necessárias:** ...
- **Métrica/KPI:** ...
- **Tipo de resposta esperada:** tabela, ranking, tendência etc.

### Respostas:

**Exemplo 1: Desempenho de Vendas por Região**
- **Pergunta de negócio:** Qual UF (estado) tem o melhor desempenho em receita e lucro? Há alguma UF com desempenho muito abaixo da média?
- **Colunas necessárias:** `uf`, `receita`, `lucro`
- **Métrica/KPI:** Receita Total por UF, Lucro Total por UF, Margem de Lucro por UF.
- **Tipo de resposta esperada:** Tabela agregada por UF, com ranking decrescente de lucro e margem.

**Exemplo 2: Rentabilidade por Produto**
- **Pergunta de negócio:** Quais produtos são os mais lucrativos para a empresa e quais podem estar com margens apertadas?
- **Colunas necessárias:** `produto`, `receita`, `lucro`
- **Métrica/KPI:** Lucro Total por Produto, Margem de Lucro por Produto.
- **Tipo de resposta esperada:** Ranking de produtos por lucro (Top N e Bottom N).

**Exemplo 3: Tendência de Vendas Sazonais**
- **Pergunta de negócio:** Como a receita e o lucro da empresa se comportam ao longo do ano? Existem meses de alta ou baixa sazonalidade que devemos planejar?
- **Colunas necessárias:** `data_venda`, `receita`, `lucro`
- **Métrica/KPI:** Receita Mensal, Lucro Mensal, Margem Mensal.
- **Tipo de resposta esperada:** Tabela de tendência mensal e um gráfico de linha mostrando a evolução da receita e do lucro ao longo do tempo.

## 14. O insight exige palavras

Segundo a aula, o mercado valoriza quem comunica decisões, não apenas quem roda código. fileciteturn5file0

### Tarefa
Escolha **duas tabelas agregadas** que você criou e escreva, para cada uma, **1 ou 2 frases de interpretação executiva**.

Exemplo de estilo:
- “O canal X concentra maior volume de receita, mas o canal Y apresenta margem superior.”
- “Recomenda-se revisar custos/expansão/priorização com base nesse resultado.”

### Interpretações:

**1. Análise por Canal (`analise_canal`):**
O canal Marketplace lidera em receita e lucro absoluto, indicando o maior volume de vendas. No entanto, o canal Online se destaca pela maior margem de lucro, sugerindo uma operação mais eficiente em termos de rentabilidade.

**2. Análise por Categoria (`analise_categoria`):**
A categoria Telefonia gera a maior receita total, mas apresenta a menor margem de lucro. Isso sugere a necessidade de revisar a estrutura de custos ou precificação para essa categoria, a fim de aumentar sua rentabilidade.

## 15. Case Varejo Brasil — Entrega final

A aula propõe quatro entregas centrais para o gestor. fileciteturn5file0

### Sua missão final
Organize o notebook para entregar:

1. **KPIs Globais**
2. **Tabelas por Dimensões Críticas**
   - por canal
   - por categoria
3. **Top 10 Produtos por Lucro**
4. **Conclusão final com pelo menos uma decisão tática**

### Importante
A conclusão final deve ser escrita em linguagem clara, direta e orientada à decisão.


## 15. Case Varejo Brasil — Entrega Final: Insights para Decisão

### 1. KPIs Globais da Operação

In [1]:
receita_total = df['receita'].sum()
lucro_total = df['lucro'].sum()
margem_lucro_percentual = (lucro_total / receita_total) * 100
ticket_medio = receita_total / df['pedido_id'].nunique()

print(f"**Receita Total:** R$ {receita_total:,.2f}")
print(f"**Lucro Total:** R$ {lucro_total:,.2f}")
print(f"**Margem de Lucro Global:** {margem_lucro_percentual:.2f}%")
print(f"**Ticket Médio Global:** R$ {ticket_medio:,.2f}")


NameError: name 'df' is not defined

### 2. Tabelas por Dimensões Críticas

#### Análise por Canal de Venda

In [2]:
analise_canal = df.groupby('canal_venda').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum'),
    quantidade_pedidos=('pedido_id', 'nunique')
).reset_index()

analise_canal['margem_calculada'] = (analise_canal['lucro_total'] / analise_canal['receita_total']) * 100
analise_canal['ticket_medio'] = analise_canal['receita_total'] / analise_canal['quantidade_pedidos']

display(analise_canal.sort_values(by='receita_total', ascending=False).style.format({
    'receita_total': 'R$ {:,.2f}',
    'lucro_total': 'R$ {:,.2f}',
    'margem_calculada': '{:,.2f}%',
    'ticket_medio': 'R$ {:,.2f}'
}))


NameError: name 'df' is not defined

#### Análise por Categoria de Produto

In [ ]:
analise_categoria = df.groupby('categoria').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum'),
    quantidade_pedidos=('pedido_id', 'nunique')
).reset_index()

analise_categoria['margem_calculada'] = (analise_categoria['lucro_total'] / analise_categoria['receita_total']) * 100
analise_categoria['ticket_medio'] = analise_categoria['receita_total'] / analise_categoria['quantidade_pedidos']

display(analise_categoria.sort_values(by='receita_total', ascending=False).style.format({
    'receita_total': 'R$ {:,.2f}',
    'lucro_total': 'R$ {:,.2f}',
    'margem_calculada': '{:,.2f}%',
    'ticket_medio': 'R$ {:,.2f}'
}))


### 3. Top 10 Produtos por Lucro

In [ ]:
analise_produto = df.groupby('produto').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum')
).reset_index()

analise_produto['margem_calculada'] = (analise_produto['lucro_total'] / analise_produto['receita_total']) * 100

print("**Top 10 Produtos por Lucro:**")
display(analise_produto.sort_values(by='lucro_total', ascending=False).head(10).style.format({
    'receita_total': 'R$ {:,.2f}',
    'lucro_total': 'R$ {:,.2f}',
    'margem_calculada': '{:,.2f}%'
}))


### 4. Conclusão Final e Decisão Tática

A análise dos dados revela que o **Marketplace** é o principal motor de receita e lucro absoluto, indicando um alto volume de vendas. No entanto, o canal **Online** demonstra uma maior eficiência em termos de margem de lucro. No que tange aos produtos, os "Smartphones" dominam em lucro, mas categorias como "Telefonia" (onde os smartphones estão inseridos) apresentam margens mais baixas.

**Decisão Tática Sugerida:**

Focar na otimização da margem do canal Marketplace e da categoria Telefonia. Isso pode ser feito através de negociações de custo com fornecedores ou revisão da estratégia de precificação para produtos de alto volume, buscando aproximar a eficiência do Marketplace e da Telefonia aos níveis observados no canal Online e em outras categorias de maior margem.

## 16. Desafio extra (opcional)

Monte uma pequena tabela executiva final com:
- Receita Total
- Lucro Total
- Margem
- Ticket Médio

e depois crie uma segunda tabela por canal com os mesmos indicadores.

Compare os resultados e explique:
- o KPI global esconde alguma diferença importante entre os canais?


In [ ]:
# Desafio extra opcional


## 17. Entrega esperada

Seu notebook deve demonstrar:
- organização
- fórmulas corretas de KPI
- uso adequado de `groupby`, `agg`, `sort_values`
- leitura temporal coerente
- interpretação escrita dos resultados

### Mensagem principal da aula
Transformar linhas de dados limpos em **apoio real para decisões de negócio**. fileciteturn5file0
